core/Dockerfile

```diff
@@ -1,3 +1,27 @@
+### Build train-sim dynamic lib (not used in core for now, but eventually will)
+# Will be used for testing
+FROM rust:latest AS train_sim_rust_env
+WORKDIR /app
+
+RUN rustup component add llvm-tools && \
+    rustup component add rustfmt && \
+    rustup component add clippy && \
+    cargo install --locked grcov
+ENV RUSTFLAGS="-Cinstrument-coverage"
+ENV LLVM_PROFILE_FILE="train-sim-%p-%m.profraw"
```

Comment:
Coverage instrumentation has a performance cost (counters must be atomically incremented for every branch and functions) - and make little sense with optimizations enabled.

Imo we don't want this for release builds.